File: newAngApp/src/Models/appointment.ts
export type AppointmentStatus = 'BOOKED' | 'RESCHEDULED' | 'CANCELLED' | 'COMPLETED' | 'NO_SHOW';

export interface Appointment {
  id: string;
  doctorId: number;
  doctorName?: string;
  patientName: string;
  appointmentDate: string; // YYYY-MM-DD
  slot: string;            // "10:30"
  status: AppointmentStatus;
}
export interface BookAppointmentRequest {
  patientId?: string;
  patientName: string;
  doctorId: number;
  specialization: string;
  appointmentDate: string; // YYYY-MM-DD
  slot: string;            // "10:30"
  reason: string;
  appointmentStatus:string;
}

export interface RescheduleAppointmentRequest {
  appointmentId: string;
  newDate: string;  // YYYY-MM-DD
  newSlot: string;  // "11:00"
  reason: string;
}

export interface CancelAppointmentRequest {
  appointmentId: string;
  reason: string;
  cancelledBy: 'PATIENT' | 'ADMIN';
}

export interface AppointmentList {
  appointmentId: number;
  slotId: number;
  doctorId: number;
  appointmentDate: string;
  startTime: string;
  endTime: string;
  patientName: string;
  doctorName: string;
  specialization: string;
  status: string;
}

export interface SlotAvailability {
  doctorId: number;
  date: string;     // YYYY-MM-DD
  slots: string[];  // ["10:00","10:30"...]
}

File: healthWebAPi/Models/Appointment.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace healthWebApi
{
    public class Appointment
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int AppointmentId { get; set; }
         public int PatientId { get; set; }
         [ForeignKey(nameof(PatientId))]
        public Patient? Patient { get; set; }
        public int DoctorId { get; set; }
        [ForeignKey(nameof(DoctorId))]
        public Doctor? Doctor { get; set; }
 public int SlotId { get; set; }
          [ForeignKey(nameof(SlotId))]

    public DoctorTimeSlot? Slot { get; set; } 
    public DateTime AppointmentDate { get; set; }  
    public TimeSpan StartTime { get; set; }
    public TimeSpan EndTime { get; set; }

    public AppointmentStatus Status { get; set; } = AppointmentStatus.Scheduled;
    public string? Reason { get; set; }
    public string? CancelReason { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime? UpdatedAt { get; set; }
public bool IsActive {get; set;}
}
}

public enum AppointmentStatus
{
    Scheduled = 1,
    Rescheduled = 2,
    Cancelled = 3
}